File: week2/day2/Pirate_exam/Pirate_exam/Pirate.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Pirate_exam
{
    class Pirate
    {
        public int drinkCount = 0;
        public string state = "Alive";
        public void DrinkSomeRum()
        {
            this.drinkCount++;
        }

        public string HowsItGoingMate()
        {
            if ((this.drinkCount >= 0) & (this.drinkCount <= 4))
            {
                return "Pour me anudder!";
            }
            else
            {
                return "Arghh, I'ma Pirate. How d'ya d'ink its goin?";
            }
        }

        public string Die()
        {
            this.state = "Dead";
            return "He's dead";
        }

        public void Brawl(Pirate x)
        {
            Random rt=new Random();
            int outcome=rt.Next(1, 4);
            if (outcome == 1)
            {
                this.Die();
            }else if (outcome == 2)
            {
                x.Die();
            }else
            {
                x.Die();
                this.Die();
            }
        }
    }
}

File: DSA-2019/week-02/day-1/pirate.py
import random


class Pirate:
    def __init__(self):
        self.rum_counter = 0
        self.is_alive = True
        self.asleep = False

    def drink_some_rum(self):
        if not self.is_alive:
            print("he is dead")
            return None
        self.rum_counter += 1

    def hows_it_going_mate(self):
        if not self.is_alive:
            print("he is dead")
            return None
        if self.rum_counter < 5:
            print("Pour me anudder!")
        else:
            self.asleep = True
            print("Arghh, I'ma Pirate. How d'ya d'ink its goin?")

    def die(self):
        self.is_alive = False
        print("he's dead")

    def brawl(self, enemy):
        # 1 means the pirate dies
        # 2 means the enemy dies
        # 3 means both die
        num = random.randint(1, 3)

        if enemy.is_alive and self.is_alive:
            if num == 1:
                self.die()
            elif num == 2:
                enemy.die()
            else:
                self.asleep = True
                enemy.asleep = True
